forwarder/internal/collections/iterator.go

```diff
@@ -26,6 +26,9 @@ func New[ReturnType any, PagerType any](
 
 	return func(yield func(ReturnType) bool) {
 		for pager.More() {
+			if ctx.Err() != nil {
```

Comment:
Hmm, while it doesn't hurt the function of the program, generally you shouldn't need to do this all over the place. I would expect `pager.NextPage` to do this under the hood and return an error. (Disclaimer that I have not checked the underlying source code there, but it's generally true as long as they're not completely batshit and using their own http client implementations or something)